_posts/2022-02-08-opensearch-java-runtime.markdown

```diff
@@ -0,0 +1,78 @@
+excerpt: ""
+
+---
+At the time of the fork OpenSearch inherited bundling OpenJDK 15, and we have made 8 releases with AdoptOpenJDK 15.0.1+9 as the default runtime, replaced with Adoptium (Temurin) 17.0.2+8 in OpenSearch 1.3.0, which is a Long-Term Support (LTS) release. LTS releases focus on stability, therefore you can expect future versions of OpenSearch to always default to bundling an LTS JDK.
+
+| Version                                                         | Bundled JDK (Linux)         |
+|:---------------------------------------------------------------:|:---------------------------:|
+| [1.0.0](https://opensearch.org/versions/opensearch-1-0-0.html)  | AdoptOpenJDK 15.0.1+9       |
+| [1.0.1](https://opensearch.org/versions/opensearch-1-0-1.html)  | AdoptOpenJDK 15.0.1+9       |
+| [1.1.0](https://opensearch.org/versions/opensearch-1-1-0.html)  | AdoptOpenJDK 15.0.1+9       |
+| [1.2.0](https://opensearch.org/versions/opensearch-1-2-0.html)  | AdoptOpenJDK 15.0.1+9       |
+| [1.2.1](https://opensearch.org/versions/opensearch-1-2-1.html)  | AdoptOpenJDK 15.0.1+9       |
+| [1.2.2](https://opensearch.org/versions/opensearch-1-2-2.html)  | AdoptOpenJDK 15.0.1+9       |
+| [1.2.3](https://opensearch.org/versions/opensearch-1-2-3.html)  | AdoptOpenJDK 15.0.1+9       |
+| [1.2.4](https://opensearch.org/versions/opensearch-1-2-4.html)  | AdoptOpenJDK 15.0.1+9       |
+| [1.3.0](https://opensearch.org/versions/opensearch-1-3-0.html)  | Adoptium (Temurin) 17.0.2+8 |
+
+This, however, doesn't tell the whole story. The OpenSearch distribution is comprised of the engine and a dozen plugins. Both the engine and each plugin is built and tested with a range of JDKs (a subset of 8, 11, 14, 15, 17, etc.), across multiple operating systems (Linux, Windows, FreeBSD, MacOS, etc.). Finally, the complete distribution is rebuilt from source, and tested with the bundled JDK.
+
+### Versions 1.0-1.2.0
+
+The complete distribution 1.0 through 1.2.4 was built with JDK 14 and tested with the bundled JDK 15. Various individual components were built and tested individually with a different combination of JDKs, mostly 14. For example, the OpenSearch engine was built with JDK 14, but tested with JDK 8, 11, 14, but not 15. Different components individually used different versions for building and testing.  
+
+### Version 1.3.0
+
+In 1.3.0 we have replaced version 14 with an LTS version 11 for builds, and are testing the complete distribution with the bundled version 17. All components build and test individually with JDK 8, 11, 14 and 17.
+
+The parent issue for this change is [opensearch-build#74](https://github.com/opensearch-project/opensearch-build/issues/74). The change in OpenSearch engine was [OpenSearch#940](https://github.com/opensearch-project/OpenSearch/pull/940), and was followed by plug-ins, e.g. [security#1580](https://github.com/opensearch-project/security/pull/1580).
+
+#### Customizing the OpenSearch Runtime
+
```

Comment:
I would also add

```
By default, any OpenSearch distribution consults `JAVA_HOME` first in order to find out the JVM runtime to run on. If `JAVA_HOME` is not set, OpenSearch will try to fallback to [bundled JVM runtime] if available.
```

Would be great to link  [bundled JVM runtime]  to the table from above.